packages/hotspot/src/hotspot/rectangle.jsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Rect, Group, Text } from 'react-konva';
+import { withStyles } from '@material-ui/core/styles';
+
+class RectComponent extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  handleClick = (e) => {
+    const { width, height, onClick, id, selected, disabled } = this.props;
+
+    if (!disabled) {
+      if (width < 0 && height < 0) {
+        return;
+      }
+      e.cancelBubble = true;
+      onClick({ id, selected: !selected });
+    }
+  };
+
+  handleMouseEnter = () => {
+    const { disabled } = this.props;
+
+    if (!disabled) {
+      document.body.style.cursor = 'pointer';
+    }
+  };
+
+  handleMouseLeave = () => {
+    document.body.style.cursor = 'default';
+  };
+
+  getEvaluateOutlineColor = (isCorrect, outlineColor) => isCorrect ? outlineColor : 'red';
+
+  getEvaluateOutlineWidth = (selected, isCorrect) => selected ? 2 : (isCorrect ? 0 : 2);
+
+  getDefaultOutlineWidth = (selected) => selected ? 2 : 0;
+
+  // TODO: Replace text with icons
+  getEvaluateText = (isCorrect, selected) => {
+    if (selected && isCorrect) {
+      return 'Correctly\nselected';
+    } else if (selected && !isCorrect) {
+      return 'Incorrectly\nselected';
+    } else if (!selected && isCorrect) {
+      return 'Correctly\nunselected';
+    }
+    return 'Incorrectly\nunselected';
+  };
+
+  render() {
+    const {
+      classes,
+      height,
+      hotspotColor,
+      isCorrect,
+      isEvaluateMode,
+      outlineColor,
+      selected,
+      width,
+      x,
+      y
+    } = this.props;
+
+    const outlineColorParsed = isEvaluateMode
+      ? this.getEvaluateOutlineColor(isCorrect, outlineColor)
+      : outlineColor;
+
+    const outlineWidth = isEvaluateMode
+      ? this.getEvaluateOutlineWidth(selected, isCorrect)
+      : this.getDefaultOutlineWidth(selected);
+
+    return (
+      <Group>
+        <Rect
+          classes={classes.base}
+          width={width}
+          height={height}
+          fill={hotspotColor}
+          onClick={this.handleClick}
+          draggable={false}
+          stroke={outlineColorParsed}
+          strokeWidth={outlineWidth}
+          onMouseLeave={this.handleMouseLeave}
+          onMouseEnter={this.handleMouseEnter}
+          x={x}
+          y={y}
+        />
+
+        {isEvaluateMode ? (
```

Comment:
you don't need 'evaluate' in your component, often just disabled and has correct response info is enough

Reply:
Not sure if I get it, are you saying that `isEvaluateMode ` can be replaced with a combination between disabled and has correct response in all occurrences from `rectangle.jsx`? If so, what if there's no correct response?

Reply:
you can leave it - the point is that the controller will prepare the ui model for you, so pie specific attributes aren't needed if you want to make your logic simpler.